test/unit/test-amp-layout.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElementWithAttributes} from '#core/dom';
+import {Layout} from '#core/dom/layout';
+
+describes.realWin('amp-layout', {amp: true}, (env) => {
+  async function getAmpLayout(attrs, innerHTML) {
+    const {win} = env;
+    const ampLayout = createElementWithAttributes(
+      win.document,
+      'amp-layout',
+      attrs
+    );
+    ampLayout.innerHTML = innerHTML;
```

Comment:
fyi, the more common way unit tests normally set up dom is with `htmlFor`: https://github.com/ampproject/amphtml/blob/d0fe752afb4c1fb24b0053fad8933b8a41da7c6a/extensions/amp-iframe/1.0/test/test-amp-iframe.js#L43-L46

Reply:
Can you speak to the benefit of htmlFor vs. innerHTML?

Reply:
Good question! It's one of those things that has a bit of that "we've always done it this way" bias, so it's good you're asking.

Component unit tests tend to prefer `htmlFor` for static DOM that's even the slightest more complicated than single node, otherwise DOM APIs are preferred next. There's a stronger guarantee over `innerHtml` in both of those approaches that the markup your inserting is well formed, plus you also get auto formatting for html blocks.

In this specific test, things could be more readable by using `htmlFor` as well:
```
const ampLayout = html `
  <amp-layout layout="fixed" width="100" height="100">
    <span>hello</span>
  </amp-layout>
`
```

I personally tend to avoid it whenever there are other tools to accomplish the same goal, and often recommend the same for others, but it's not a hard and fast rule. This is more critical in production code particularly in cases where XSS or performance may be a concern, but here it's more of a stylistic choice.